helix-view/src/document.rs

```diff
@@ -457,6 +459,106 @@ where
     *mut_ref = f(mem::take(mut_ref));
 }
 
+/// Trait representing ways of configuring the document as it's being opened.
+trait ConfigureDocument {
+    type Config;
+    /// Loads document configuration for a file at `path`.
+    fn load(&self, path: &Path) -> Result<Self::Config, Error>;
+    /// Retrieves the encoding from a `Config`.
+    fn encoding(config: &Self::Config) -> Option<&'static encoding::Encoding>;
+    /// Retrieves the line ending from a `Config`.
+    fn line_ending(config: &Self::Config) -> Option<LineEnding>;
+    /// Applies any document configuration not handled by one of the other methods.
+    fn configure_document(doc: &mut Document, config: Self::Config);
+}
+
+/// Document configuration strategy that uses fallback auto-detection as a first resort.
+#[derive(Clone, Copy, Debug, Default)]
+struct Autodetect;
+/// Document configuration strategy that loads configuration from `.editorconfig` files.
+#[derive(Clone, Copy, Debug, Default)]
+struct EditorConfig;
+
+impl ConfigureDocument for Autodetect {
+    type Config = ();
```

Comment:
I don't see how this should be an associate trait, these data could be put within the struct itself.